tabular/src/autogluon/tabular/models/tabpfnmix/tabpfnmix_model.py

```diff
@@ -0,0 +1,221 @@
+        if self.problem_type in ['binary', 'multiclass']:
+            model_cls = TabPFNMixClassifier
+        else:
+            # FIXME: Add regression support
+            raise AssertionError(f"TabPFN does not support problem_type='{self.problem_type}'")
+        return model_cls
+
+    def _set_default_params(self):
+        """Specifies hyperparameter values to use by default"""
+        default_params = {
+            "n_ensembles": 1,
+            "max_epochs": 0,
+            "split_val": False,
+            "use_best_epoch": True,
+        }
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    # FIXME: Make X_val, y_val = None work well, currently it uses X and y as validation, should instead skip validation entirely
+    # FIXME: Use `params_trained` for refit optimal epochs
+    # FIXME: Handle model weights download
+    # FIXME: GPU support?
+    # FIXME: Save model weights to file instead of pickling?
+    def _fit(self, X: pd.DataFrame, y: pd.Series, X_val: pd.DataFrame = None, y_val: pd.Series = None, num_cpus: int = 1, num_gpus: float = 0, **kwargs):
+        try_import_torch()
+        import torch
+        from ._internal.config.config_run import ConfigRun
+
+        ag_params = self._get_ag_params()
+        max_classes = ag_params.get("max_classes")
+        if max_classes is not None and self.num_classes > max_classes:
+            # TODO: Move to earlier stage when problem_type is checked
+            raise AssertionError(f"Max allowed classes for the model is {max_classes}, " f"but found {self.num_classes} classes.")
+
+        params = self._get_model_params()
+
+        # FIXME: Don't require loading from file, allow user to specify everything?
+        # TODO: Not a big fan of the path config logic due to possible portability issues
+        if "path_config" in params:
+            path_config = Path(params["path_config"])
+        else:
+            dir_path = os.path.dirname(os.path.realpath(__file__))
+            path_config = Path(dir_path) / "configs" / "tabpfnmix_base.yaml"
+        cfg = ConfigRun.load(Path(path_config))
+        if "path_weights" not in params:
+            raise ValueError(
+                "Missing required model hyperparameter 'path_weights'. "
+                "Either specify `path_weights=None` to train from random initialization (not recommended), "
+                "or specify a local path to a pre-trained weights file such as `path/to/file/tabpfnmix_base.pt`."
+            )
+
+        from ._internal.core.enums import Task
+        if self.problem_type in [REGRESSION, QUANTILE]:
+            task = Task.CLASSIFICATION
+        else:
+            task = Task.REGRESSION
+        cfg.task = task
+
+        if cfg.device is None:
+            if num_gpus > 0:
+                # TODO: Test with GPU
+                # TODO: What if multi-gpu?
+                cfg.device = 'cuda:0'
```

Comment:
Is it expected to use only 1 GPU for fine-tuning in case of multiple available?

Reply:
Currently I haven't tested GPU support. I don't know what will happen when the user fits with GPU. I lack bandwidth so if someone wants to try it out with GPU, they can let me know how it goes.